src/cephadm/cephadm

```diff
@@ -2268,12 +2271,17 @@ def list_daemons(detail=True, legacy_dir=None):
                         if not code:
                             (container_id, image_name, image_id, version) = out.strip().split(',')
                             image_id = normalize_container_id(image_id)
-                            if not version:
-                                out, err, code = call(
-                                    [container_path, 'exec', container_id,
-                                     'ceph', '-v'])
-                                if not code and out.startswith('ceph version '):
-                                    version = out.split(' ')[2]
+                            if name.split('.', 1)[0] not in Monitoring.components:
```

Comment:
Maybe invert this logic to make the branch specific to ceph and then add a `FIXME` for monitoring?

something like  ... ?
```suggestion
                            daemon_type = name.split('.', 1)[0]
                            if daemon type in Ceph.components:
                                # current `ceph -v` logic
                            elif daemon_type in Monitoring.components:
                               # FIXME: implement version check for monitoring components
                               pass
                           
```